spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -100,5 +83,22 @@ module.exports = function (provider, setup) {
         done();
       });
   });
+
+  it("Works as a Client", function (done) {
+    socket.connect('www.google.com', 80, function () {
+      console.warn('connected');
+      var written = false;
+      socket.write(rawStringToBuffer('GET / HTTP/1.0\n\n'), function (okay) {
+        console.warn('written');
+        written = true;
+      });
+      setTimeout(function () {
```

Comment:
instead, we should not have setTImeouts, but rather have the events happen as things happen. so in the callback to socket.write, we should expect less then 500ms to have passed since it was called.  We should add an onData listener which also expect's to be called, and done() should be moved into the callback of socket.close().

I think restructuring that way would remove flakiness